node/src/Pos/Client/Txp/Util.hs

```diff
@@ -285,11 +314,18 @@ prepareTxRaw utxo outputs (TxFee fee) = do
             let trOutputs = outputs
             pure TxRaw {..}
   where
+    onlyConfirmedInputs :: S.Set Address -> TxOutAux -> Bool
+    onlyConfirmedInputs addrs (TxOutAux (TxOut addr _)) = not (addr `S.member` addrs)
+
     sumTxOuts = either (throwError . GeneralTxError) pure .
         integerToCoin . sumTxOutCoins
-    allUnspent = M.toList utxo
+    -- Try to filter only "confirmed" addresses, defaulting to the old policy if the
+    -- filtered collection is empty.
+    allUnspent = case M.filter (onlyConfirmedInputs (allPendingAddresses pendingTx)) utxo of
+        x | M.null x -> utxo -- pass the full, unfiltered utxo.
+        x -> x
```

Comment:
As @fersel points out, this has a corner case.

Here's another suggestion:

    sortedUnspent = confirmed ++ unconfirmed
      where
        (confirmed, unconfirmed) =
          partition (onlyConfirmedInputs (allPendingAddresses pendingTx))
                    (sortOn (Down . txOutValue . toaOut . snd) (M.toList utxo))